public/components/FeatureAnywhereContextMenu/AssociatedDetectors/components/ConfirmUnlinkDetectorModal/ConfirmUnlinkDetectorModal.tsx

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState } from 'react';
+import {
+  EuiText,
+  EuiOverlayMask,
+  EuiButton,
+  EuiButtonEmpty,
+  EuiModal,
+  EuiModalHeader,
+  EuiModalFooter,
+  EuiModalBody,
+  EuiModalHeaderTitle,
+} from '@elastic/eui';
+import { DetectorListItem } from '../../../../../models/interfaces';
+import { EuiSpacer } from '@elastic/eui';
+
+interface ConfirmUnlinkDetectorModalProps {
+  detector: DetectorListItem;
+  onUnlinkDetector(): void;
+  onHide(): void;
+  onConfirm(): void;
+  isListLoading: boolean;
+}
+
+export const ConfirmUnlinkDetectorModal = (
+  props: ConfirmUnlinkDetectorModalProps
+) => {
+  const [isModalLoading, setIsModalLoading] = useState<boolean>(false);
+  const isLoading = isModalLoading || props.isListLoading;
+  return (
+    <EuiOverlayMask>
+      <EuiModal
+        data-test-subj="startDetectorsModal"
+        onClose={props.onHide}
+        maxWidth={450}
+      >
+        <EuiModalHeader>
+          <EuiModalHeaderTitle>
+            {'Remove association?'}&nbsp;
+          </EuiModalHeaderTitle>
+        </EuiModalHeader>
+        <EuiModalBody>
+          <EuiText>
+            Removing association unlinks {props.detector.name} detector from the
+            visualization but does not delete it. The detector association can
+            be restored.
+          </EuiText>
+          <EuiSpacer size="s" />
+        </EuiModalBody>
+        <EuiModalFooter>
+          {isLoading ? null : (
+            <EuiButtonEmpty
+              data-test-subj="cancelButton"
+              onClick={props.onHide}
+            >
+              Cancel
+            </EuiButtonEmpty>
+          )}
```

Comment:
nit - could we change the loading state to just set `isLoading={true}` for the button, instead of having no button at all? In other words, remove the conditional and just populate the button's `isLoading` prop with the value of the existing `isLoading` prop that you are passing in.

Reply:
Got it - my initial thought was this would be in loading state when opening it, instead of closing/finishing the action. Makes sense to me to leave as-is.